### DISCUSSION TITLE
Question regarding on how Apple rules propagate configuration/targets to `cc_library` or `objc_library`

### DISCUSSION THREAD

**Discussion by Winnaries** [bazelbuild/rules_apple#1596] [Q&A]
With the release of `@platforms`, I can change targets of `cc_library` easily to `arm64` by defining a set of constraints and using `select()` on those values. However, since Apple rules doesn't support `@platforms` and instead pass the config to C build configuration through `--cpu` and `--crosstool_top`. Being new to bazel, I am confused on how things operate over the boundary. Would be very helpful if someone can answer the following questions for me: 

1. How does the Apple rules pass configuration to the underlying dependencies such as `cc_library` and `objc_library`? For example, if I want to build an XCFramework targeting `x86_64` and `arm64` that depends on `cc_library` target, how does the C target know which target to build and which toolchain to use under the absense of `@platforms`. 
2. If I have a `cc_library` rule that uses `select()` on `@platforms` (eg. emiting TARGET_IOS local defines if the target is `@platforms//os:ios`) and have a `apple_static_xcframework` rules depends on it, what would be the result of the select? From my understanding, it would resolve to default condition, right? If that is the case, should I instead make `select()` on `cpu` for compatibility with Apple rules?
3. What would be the current best way to build Apple multi-arch library? Should I stick to using `--cpu`, or `--platforms`?
4. Lastly, will Apple rules adopt `@platforms` any time soon? I would be happy to help under some guidance.

**Comment by thii** [comment]
1. Transitions. For example, the XCFramework rules use this to pass down the target environment, etc. to the dependencies https://github.com/bazelbuild/rules_apple/blob/77155d9f82996d95e1f6cd944e4acbe5cd637f59/apple/internal/transition_support.bzl#L566-L591
2. I don't entirely understand the question, but if you build an `apple_static_xcframework` for multiple platforms, and use `select()` on `@platforms//os:ios` for part of the deps, those should be built only when Bazel is building them for the iOS platform. Please file any issues you find.
3. Building with `--platforms` is not ready yet, I believe the current best way is still using `--cpu`.
4. Platform support for Apple rules is in-progress.
